proxy/src/main/java/com/wavefront/agent/preprocessor/PreprocessorUtil.java

```diff
@@ -261,34 +262,34 @@ private static Predicate processComparisonOp(Map.Entry<String, Object> subElemen
     if (reportableEntity == ReportPoint.class) {
       switch (subElement.getKey()) {
         case "equals":
-          return new ReportPointEqualsPredicate(scope, ruleVal);
+          return new ReportPointEqualsPredicate(scope, (String) ruleVal);
         case "startsWith":
-          return new ReportPointStartsWithPredicate(scope, ruleVal);
+          return new ReportPointStartsWithPredicate(scope, (String) ruleVal);
         case "contains":
-          return new ReportPointContainsPredicate(scope, ruleVal);
+          return new ReportPointContainsPredicate(scope, (String) ruleVal);
         case "endsWith":
-          return new ReportPointEndsWithPredicate(scope, ruleVal);
+          return new ReportPointEndsWithPredicate(scope, (String) ruleVal);
         case "regexMatch":
-          return new ReportPointRegexMatchPredicate(scope, ruleVal);
+          return new ReportPointRegexMatchPredicate(scope, (String) ruleVal);
         case "in":
-          return new ReportPointInPredicate(scope, ruleVal);
+          return new ReportPointInPredicate(scope, (ArrayList) ruleVal);
```

Comment:
List<String>. In this context specific implementation (ArrayList) is not important but the element's type kind of is.